File: frontend/src/modules/admin/types/securityLog.ts
/**
 * 安全日志相关类型定义
 */

// =============================================================================
// 事件类型和严重程度
// =============================================================================

export type SecurityEventType =
  | 'login_failed'
  | 'login_anomaly'
  | 'request_error'
  | 'request_blocked'
  | 'service_error'
  | 'service_recovered'
  | 'daily_report'
  | 'config_changed';

export type SecuritySeverity = 'info' | 'warning' | 'error' | 'critical';

// 事件类型显示名称
export const EVENT_TYPE_DISPLAY: Record<SecurityEventType, string> = {
  login_failed: '登录失败',
  login_anomaly: '异常登录',
  request_error: '请求异常',
  request_blocked: '请求拦截',
  service_error: '服务异常',
  service_recovered: '服务恢复',
  daily_report: '每日报告',
  config_changed: '配置变更',
};

// 严重程度显示名称
export const SEVERITY_DISPLAY: Record<SecuritySeverity, string> = {
  info: '信息',
  warning: '警告',
  error: '错误',
  critical: '严重',
};

// 严重程度颜色
export const SEVERITY_COLORS: Record<SecuritySeverity, string> = {
  info: 'text-blue-600 bg-blue-50 border-blue-200',
  warning: 'text-orange-600 bg-orange-50 border-orange-200',
  error: 'text-red-600 bg-red-50 border-red-200',
  critical: 'text-red-800 bg-red-100 border-red-300',
};

// =============================================================================
// 日志数据结构
// =============================================================================

export interface SecurityLogItem {
  id: string;
  event_type: SecurityEventType;
  severity: SecuritySeverity;
  title: string;
  description: string;
  ip_address: string | null;
  user_id: string | null;
  username: string | null;
  extra_data: Record<string, unknown>;
  archived: boolean;
  created_at: string;
}

export interface SecurityLogGroup {
  date: string;
  date_display: string;
  logs: SecurityLogItem[];
  count: number;
}

// =============================================================================
// API 响应类型
// =============================================================================

export interface SecurityLogListResponse {
  groups: SecurityLogGroup[];
  total: number;
  has_more: boolean;
}

export interface SecurityLogStatsResponse {
  total_count: number;
  error_count: number;
  warning_count: number;
  info_count: number;
  last_error_at: string | null;
  last_daily_report_at: string | null;
}

export interface SecurityLogExportResponse {
  filename: string;
  content: string; // Base64 编码
  content_type: string;
  record_count: number;
}

export interface SecurityLogDeleteResponse {
  deleted_count: number;
}

export interface SecurityLogArchiveResponse {
  archived_count: number;
}

// =============================================================================
// 请求参数类型
// =============================================================================

export interface SecurityLogQueryParams {
  event_types?: SecurityEventType[];
  severities?: SecuritySeverity[];
  start_date?: string;
  end_date?: string;
  include_archived?: boolean;
  page?: number;
  page_size?: number;
}

export interface SecurityLogDeleteRequest {
  log_ids?: string[];
  before_date?: string;
  delete_all?: boolean;
}

export interface SecurityLogExportRequest {
  format?: 'json' | 'csv';
  event_types?: SecurityEventType[];
  severities?: SecuritySeverity[];
  start_date?: string;
  end_date?: string;
  include_archived?: boolean;
}

export interface SecurityLogArchiveRequest {
  before_date: string;
}

File: backend/app/api/v1/schemas/security_log.py
"""
安全日志 API Schema

定义安全日志相关的请求和响应模型
"""

from datetime import datetime

from pydantic import BaseModel, Field

from app.domain.models.security_log import SecurityEventType, SecuritySeverity

# =============================================================================
# 响应模型
# =============================================================================


class SecurityLogItem(BaseModel):
    """单条安全日志"""

    id: str
    event_type: SecurityEventType
    severity: SecuritySeverity
    title: str
    description: str
    ip_address: str | None = None
    user_id: str | None = None
    username: str | None = None
    extra_data: dict = Field(default_factory=dict)
    archived: bool = False
    created_at: datetime

    class Config:
        from_attributes = True


class SecurityLogGroup(BaseModel):
    """按日期分组的安全日志"""

    date: str  # YYYY-MM-DD
    date_display: str  # 显示名称，如"今天"、"昨天"、"2026-01-25"
    logs: list[SecurityLogItem]
    count: int


class SecurityLogListResponse(BaseModel):
    """安全日志列表响应"""

    groups: list[SecurityLogGroup]
    total: int
    has_more: bool


class SecurityLogStatsResponse(BaseModel):
    """安全日志统计响应"""

    total_count: int
    error_count: int
    warning_count: int
    info_count: int
    last_error_at: datetime | None = None
    last_daily_report_at: datetime | None = None


# =============================================================================
# 请求模型
# =============================================================================


class SecurityLogQueryParams(BaseModel):
    """安全日志查询参数"""

    event_types: list[SecurityEventType] | None = None
    severities: list[SecuritySeverity] | None = None
    start_date: datetime | None = None
    end_date: datetime | None = None
    include_archived: bool = False
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=50, ge=1, le=200)


class SecurityLogDeleteRequest(BaseModel):
    """安全日志删除请求"""

    log_ids: list[str] | None = None  # 指定 ID 删除
    before_date: datetime | None = None  # 删除指定日期之前的日志
    delete_all: bool = False  # 删除所有（需要确认）


class SecurityLogExportRequest(BaseModel):
    """安全日志导出请求"""

    format: str = Field(default="json", pattern="^(json|csv)$")
    event_types: list[SecurityEventType] | None = None
    severities: list[SecuritySeverity] | None = None
    start_date: datetime | None = None
    end_date: datetime | None = None
    include_archived: bool = False


class SecurityLogExportResponse(BaseModel):
    """安全日志导出响应"""

    filename: str
    content: str  # Base64 编码的文件内容
    content_type: str
    record_count: int


class SecurityLogArchiveRequest(BaseModel):
    """安全日志归档请求"""

    before_date: datetime  # 归档指定日期之前的日志


class SecurityLogArchiveResponse(BaseModel):
    """安全日志归档响应"""

    archived_count: int
